src/lib/consensus/proof_of_stake.ml

```diff
@@ -1876,6 +1880,52 @@ module Data = struct
 
     let is_genesis_state_var (t : var) = is_genesis t.curr_global_slot
 
+    let supercharge_coinbase
+        ~(constraint_constants : Genesis_constants.Constraint_constants.t)
+        ~(delegator_account : Coda_base.Account.var) ~ledger ~global_slot =
+      let open Snark_params.Tick in
+      (*let%bind winner_addr =
+          request_witness
+            (Coda_base.Account.Index.Unpacked.typ
+               ~ledger_depth:constraint_constants.ledger_depth)
+            (As_prover.return Winner_address)
+        in*)
+      let%bind public_key =
+        request_witness Public_key.typ (As_prover.return Vrf.Public_key)
+      in
+      (*let%bind delegator_account =
+          with_label __LOC__
+            (Coda_base.Frozen_ledger_hash.get
+               ~depth:constraint_constants.ledger_depth ledger winner_addr)
+        in*)
+      let%bind delegate_account =
+        let delegate_account_id =
+          Coda_base.Account_id.Checked.create delegator_account.delegate
+            Coda_base.Token_id.(var_of_t default)
+        in
+        let%bind delegate_addr =
```

Comment:
We should be passing the account in from the vrf evaluation rather than re-existing it and calling `get`.

Reply:
We don't look up the delegate account in vrf evaluation. We look up the delegator account which I'm passing it here

Reply:
So given the conversation above, we only need the delegator, which we already looked up right?

Reply:
yup, Just pushed that change